tests/_multi_screen.lua

```diff
@@ -0,0 +1,509 @@
+-- Helper module to replicate a set of steps across multiple screen scenarios
+-- Ideas for improvments:
+--  * Export an iterator so the suits can call the screen change step by step
+--    themselves instead of the current all or nothing
+--  * Offer multiple scenarios such as screen add, remove, resize/swap instead
+--    of having to test all corner cases all the time.
+local wibox = require("wibox")
+local surface = require("gears.surface")
+local wallpaper = require("gears.wallpaper")
+local color = require("gears.color")
+local shape = require("gears.shape")
+
+local module = {}
+
+-- Get the original canvas size before it is "lost"
+local canvas_w, canvas_h = screen[1].geometry.width, screen[1].geometry.height
+
+-- Create virtual screens.
+-- The steps will be executed on each set of screens.
+local dispositions = {
```

Comment:
It is what I did at first. Then I imagined a future improvements where there is multiple scenarios for different unit tests. For example, some tests only care about the number of screen changing, they don't need the full disposition testing (it is quite long and slow).

To push things further, then functions are required, as the same could be used in multiple "stages"* and change its output based on the current state. It would save quite a lot of code. The functions are currently necessary, but this will change. If I ever finish Tyrannical tests**, I will need this.

\* A screen configuration change.
\** I just `ln -s` test-tyrannical-screen.lua and test-tyrannical-rules.lua into Awesome `awesome/tests/` folder and `ln -s` tyrannical itself into `awesome/lib` locally.